tests/test_conversation.py

```diff
@@ -0,0 +1,127 @@
+        messages = [
+            database.Message(
+                filename="1-message.gpg",
+                is_downloaded=True,
+                content="Hello! I think this is newsworthy: ...",
+            ),
+            database.Message(
+                filename="6-message.gpg",
+                is_downloaded=True,
+                content="I can send you more if you're interested.",
+            ),
+            database.Message(
+                filename="5-message.gpg",
+                is_downloaded=True,
+                content="Here is a document with details!",
+            ),
+            database.Message(
+                filename="8-message.gpg",
+                is_downloaded=True,
+                content="Sure.",
+            ),
+        ]
+        interested_journalist = database.User(username="interested-journalist")
+        other_journalist = database.User(username="other-journalist")
+        replies = [
+            database.Reply(
+                journalist=interested_journalist,
+                filename="2-reply.gpg",
+                is_downloaded=True,
+                content=dedent(
+                    """\
+                    Thank you for the tip!
+
+                    Can you tell me more about... ?
+                    """
+                ),
+            ),
+            database.Reply(
+                journalist=interested_journalist,
+                filename="3-reply.gpg",
+                is_downloaded=True,
+                content=dedent(
+                    """\
+                    Do you have proof of...?
+                    """
+                ),
+            ),
+            database.Reply(
+                journalist=other_journalist,
+                filename="7-reply.gpg",
+                is_downloaded=True,
+                content=dedent("Yes, the document you sent was useful, I'd love to see more."),
+            ),
+        ]
+        draft_reply = database.DraftReply(
+            content="Let me think...",
+            file_counter=2,
+            timestamp=datetime.now(),
+        )
+        source.files = files
+        source.messages = messages
+        source.replies = replies
+        source.draftreplies = [draft_reply]
+
+        self._source = source
+
+    def test_indicates_explicitly_absence_of_messages(self):
+        source = database.Source()
+        assert str(conversation.Transcript(source)) == "No messages."
+
+    def test_renders_all_messages(self):
+        assert str(conversation.Transcript(self._source)) == dedent(
+            """\
+            happy-bird wrote:
+            Hello! I think this is newsworthy: ...
+            ------
+            interested-journalist wrote:
+            Thank you for the tip!
+
+            Can you tell me more about... ?
+
+            ------
+            Do you have proof of...?
+
+            ------
+            happy-bird sent:
+            File: 4-memo.pdf.gpg
+            ------
+            happy-bird wrote:
```

Comment:
Note that this context statement wasn't in the initial proposal, but adding it simplifies the template at this stage (by allowing the context strings to be treated as opaque strings). Once all the template management is performed in `conversation.Transcript`, we'll be able to do whatever we want without extra complexity.